### PR TITLE
Issue #13345: Enabled PatternVariableNameCheckExamplesTest

### PR DESCRIPTION
**Pull request by MANISH-K-07**
Part of #13345 

Enabled PatternVariableNameCheckExamplesTest

### CODE REVIEW

**Comment by MANISH-K-07** [comment]
I think there is some issue with the CircleCI workflow ?....
The failing checks seem irrelevant to the changes I have made

**Comment by MANISH-K-07** [comment]
There seems to be some sort of issue with CircleCI workflow ?....
The failures seem irrelevant to my changes

**Comment by MANISH-K-07** [comment]
GitHub, generate website

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class Example2 {
+  Example2(Object o1){
+    if (o1 instanceof String STR) { // violation
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java:17]: Please make all inputs to be the same by java code and different only in comments

**Review by romani — CHANGES_REQUESTED** [review]
Thanks a lot, one item to make it ideal:

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class Example2 {
+  Example2(Object o1){
+    if (o1 instanceof String STRING) { // violation
+    }
+    if (o1 instanceof Integer num) {
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java:19]: Please add to all inputs `num_1` to show that no violation is reported if allowed in format

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java (hunk 2 above)
Comment (MANISH-K-07) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java:19]: Yep Done....

**Comment by MANISH-K-07** [comment]
GitHub, generate website

**Comment by MANISH-K-07** [comment]
GitHub, generate website

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class Example1 {
+  Example1(Object o1){
+    if (o1 instanceof String STRING) { // violation
+    }
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java:16]: please make code like `if (o1 instanceof String STRING) {} // violation`
atention to `{}`  to let user not bother wtih block at all.

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class Example1 {
+  Example1(Object o1){
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java:14]: `foo(Object o1){`

to minimize diff between exmaples.

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example3.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class Example3 {
+  Example3(Object o1){
+    if (o1 instanceof String STRING) { // violation
+    }
+    if (o1 instanceof Integer num) {
+    }
+    if (o1 instanceof Integer num_1) {
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example3.java:21]: please make some change in this example to make it different from Example2 by violations.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example3.java (hunk 5 above)
Comment (MANISH-K-07) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example3.java:21]: Added new set of violation and made all the examples same by java code.
Please review

**Comment by MANISH-K-07** [comment]
Github, generate website

**Comment by MANISH-K-07** [comment]
Github, generate website

**Comment by romani** [comment]
Github, generate website

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java (hunk 4 above)
Comment (romani) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java:14]: no done, please do

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class Example2 {
+  Example2(Object o1){
+    if (o1 instanceof String STRING) {} // violation
+
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java:18]: please remove empty lines to keep example compact.

**Review by romani — CHANGES_REQUESTED** [review]
one item is not done, see above.

items:

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java (hunk 4 above)
Comment (MANISH-K-07) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java]: `Example1(Object o1){`  is a constructor.

To change to `foo` , I had to change the class name 
and also had to add explicit class Example1 to satisfy the exampleClassName check.

Is the change okay ?

**Comment by MANISH-K-07** [comment]
Github, generate website

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class foo {
+  foo(Object o1){
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java:14]: Please do not use ctor of class, keep class as ExampleX and use regular method .

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java (hunk 4 above)
Comment (romani) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java]: Change constructor to method

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java (hunk 7 above)
Comment (MANISH-K-07) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java]: > Please do not use ctor of class, keep class as ExampleX and use regular method .

Reverted....

**Comment by MANISH-K-07** [comment]
Github, generate website

**Review by romani — APPROVED** [review]
Ok to merge.

Thanks a lot

**Pull request merged**